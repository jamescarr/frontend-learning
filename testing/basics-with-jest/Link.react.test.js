import React from 'react';
import Link from './Link.react';
import renderer from 'react-test-renderer';


// This test generates a snapshot in __snapshots__/Link.react.test.js.snap
// that contains the output from the first test run, then compares it 
// next time the test is ran. If the output changes, the test will fail.
// Made it fail by changing mouseLeave to set the state to HOVERED.
test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <Link page="http://zapier.com">Zapier Makes You Happier</Link>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // trigger the callback
    tree.props.onMouseEnter();

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the leave
    tree.props.onMouseLeave();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
