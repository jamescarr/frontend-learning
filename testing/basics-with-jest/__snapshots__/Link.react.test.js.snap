// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://zapier.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Zapier Makes You Happier
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://zapier.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Zapier Makes You Happier
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://zapier.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Zapier Makes You Happier
</a>
`;
